react/features/desktop-picker/components/DesktopPicker.js

```diff
@@ -162,7 +155,10 @@ class DesktopPicker extends Component {
      */
     componentWillUnmount() {
         this._stopPolling();
-        this.props.dispatch(resetDesktopSources());
+        this.setState({
+            sources: {},
+            types: []
+        });
```

Comment:
Is it really necessary to clear a component's state that's going to destroy itself? Are the references to the sources/types being stored somewhere else that would prevent cleanup?

Reply:
I remember hearing something that unmounting is not necessary equal to destroying the component and that's why I put the reset. But now I check the documentation and did some testing and it seems that you are correct. I'll remove the code!